ome/datasource_baseline_compliance_report.go

```diff
@@ -5,111 +5,115 @@ import (
 	"terraform-provider-ome/clients"
 	"terraform-provider-ome/models"
 
-	"github.com/hashicorp/terraform-plugin-framework/diag"
-	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-type configurationReportDataSourceType struct{}
+var (
+	_ datasource.DataSource              = &configurationReportDataSource{}
+	_ datasource.DataSourceWithConfigure = &configurationReportDataSource{}
+)
+
+// NewConfigurationReportDataSource is a new datasource for configuration report
+func NewConfigurationReportDataSource() datasource.DataSource {
+	return &configurationReportDataSource{}
+}
+
+type configurationReportDataSource struct {
+	p *omeProvider
+}
+
+// Configure implements datasource.DataSourceWithConfigure
+func (g *configurationReportDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	g.p = req.ProviderData.(*omeProvider)
+}
 
-func (t configurationReportDataSourceType) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
-	return tfsdk.Schema{
+// Metadata implements datasource.DataSource
+func (*configurationReportDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "configuration_report_info"
+}
+
+func (*configurationReportDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
 		MarkdownDescription: "Data source to list the compliance configuration report of a baseline from OpenManage Enterprise.",
-		Attributes: map[string]tfsdk.Attribute{
+		Attributes: map[string]schema.Attribute{
 
-			"id": {
+			"id": schema.StringAttribute{
 				MarkdownDescription: "ID for data source.",
 				Description:         "ID for data source.",
-				Type:                types.StringType,
 				Computed:            true,
 				Optional:            true,
 			},
-			"baseline_name": {
+			"baseline_name": schema.StringAttribute{
 				MarkdownDescription: "Name of the Baseline.",
 				Description:         "Name of the Baseline.",
-				Type:                types.StringType,
 				Required:            true,
 			},
-			"fetch_attributes": {
+			"fetch_attributes": schema.BoolAttribute{
 				MarkdownDescription: "Fetch  device compliance attribute report.",
 				Description:         "Fetch  device compliance attribute report.",
-				Type:                types.BoolType,
 				Computed:            true,
 				Optional:            true,
 			},
-			"compliance_report_device": {
+			"compliance_report_device": schema.ListNestedAttribute{
```

Comment:
complaince -> compliance in the Description and MarkDownDescription

Reply:
refactor the complaince to compliance in new commit
add the description to all the ID schema instance in new commit
[added ID description to schema](https://github.com/dell/terraform-provider-ome/pull/37/commits/6246ee46a9551848b74a2f99bd10beb899dd5151)